sympy/core/relational.py

```diff
@@ -217,7 +217,68 @@ def _eval_simplify(self, ratio, measure, rational, inverse):
                 v = S.Zero
             if v is not None:
                 r = r.func._eval_relation(v, S.Zero)
-
+            r = r.canonical
+            # If there is only one symbol in the expression,
+            # try to write it on a simplified form
+            free = r.free_symbols
+            if len(free) == 1:
+                try:
+                    from sympy.solvers.solveset import linear_coeffs
+                    x = free.pop()
+                    dif = r.lhs - r.rhs
+                    m, b = linear_coeffs(dif, x)
+                    if m.is_zero is False:
+                        if m.is_negative:
+                            # Dividing with a negative number, so change order of arguments
+                            # canonical will put the symbol back on the lhs later
+                            r = r.func(-b/m, x)
+                        else:
+                            r = r.func(x, -b/m)
+                    else:
+                        r = r.func(b, S.zero)
+                except ValueError:
+                    # maybe not a linear function, try polynomial
+                    from sympy.polys import Poly, poly, PolynomialError, gcd
+                    try:
+                        p = poly(dif, x)
+                        c = p.all_coeffs()
+                        leading = c[0]
+                        constant = c[-1]
+                        c[-1] = 0
+                        scale = gcd(c)
+                        c = [ctmp/scale for ctmp in c]
+                        r = r.func(Poly.from_list(c, x).as_expr(), -constant/scale)
+                    except PolynomialError:
+                        pass
+            elif len(free) >= 2:
+                try:
+                    from sympy.solvers.solveset import linear_coeffs
+                    from sympy.polys import gcd
+                    free = list(ordered(free))
+                    dif = r.lhs - r.rhs
+                    m = linear_coeffs(dif, *free)
+                    constant = m[-1]
```

Comment:
`constant = m.pop()` is a common idiom for these two steps

Reply:
Thanks! Seemed like there should be a better way.